warp10/src/main/java/io/warp10/script/functions/PIVOT.java

```diff
@@ -45,19 +46,28 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     Object top = stack.pop();
     
     if (!(top instanceof List) || (0 == ((List) top).size())) {
-      throw new WarpScriptException(getName() + " expects a non empty list of labeling Geo Time Series™.");
+      throw new WarpScriptException(getName() + " expects a non empty list of labeling Geo Time Series.");
     }
     
-    List<Object> labeling = (List<Object>) top;
+    List<Object> paramLabeling = (List<Object>) top;
+    List<Object> labeling = new ArrayList<Object>(paramLabeling.size());
     
     Set<String> classes = new HashSet<String>();
-    for (Object o: labeling) {
+    for (Object o: paramLabeling) {
+      if (o instanceof GTSEncoder) {
+        GTSEncoder encoder = (GTSEncoder) o;
+        GTSDecoder decoder = encoder.getDecoder(true);
+        o = decoder.decode(null, true);
```

Comment:
Catch the RuntimeException to display a nice error message.

Reply:
I agree. It will could be usefull for other functions too.